File: Week 6/Sentimental/Readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    L = count_letters(text) / count_words(text) * 100
    S = count_sentences(text) / count_words(text) * 100
    grade = round(0.0588 * L - 0.296 * S - 15.8)
    if grade > 16:
        print("Grade 16+\n")

    elif grade > 0:
        print(f"Grade {grade}\n")

    else:
        print("Before Grade 1\n")


def count_letters(text):
    ans = 0
    n = len(text)
    for i in range(n):
        if text[i].isalpha():
            ans += 1
    return ans


def count_words(text):
    ans = 0
    n = len(text)
    for i in range(n):
        if text[i] == ' ':
            ans += 1
    return ans + 1


def count_sentences(text):
    ans = 0
    n = len(text)
    for i in range(n):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            ans += 1

    return ans


main()

File: Week 2/Readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text)
{
    int ans = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            ans++;
        }
    }

    return ans;
}

int count_words(string text)
{
    int ans = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            ans++;
        }
    }

    return ans + 1;
}

int count_sentences(string text)
{
    int ans = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            ans++;
        }
    }

    return ans;
}

int main(void)
{
    string text = get_string("Text: ");

    float L = (float)count_letters(text) / count_words(text) * 100, S = (float)count_sentences(text) / count_words(text) * 100.0;
    int grade = round(0.0588 * L - 0.296 * S - 15.8);
    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade > 0)
    {
        printf("Grade %i\n", grade);
    }
    else
    {
        printf("Before Grade 1\n");
    }
}